independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/multipart/FileUpload.java

```diff
@@ -0,0 +1,44 @@
+package org.jboss.resteasy.reactive.multipart;
+
+import java.nio.file.Path;
+
+/**
+ * Represents a file-upload from an HTTP multipart form submission.
+ */
+public interface FileUpload {
+
+    /**
+     * @return the name of the upload as provided in the form submission.
+     */
+    String name();
+
+    /**
+     * @return the actual temporary file name on the server where the file was uploaded to.
+     */
+    Path uploadedFile();
+
+    /**
+     * @return the file name of the upload as provided in the form submission.
+     */
+    String fileName();
+
+    /**
+     * @return the size of the upload, in bytes
+     */
+    long size();
+
+    /**
+     * @return the content type (MIME type) of the upload.
+     */
+    String contentType();
+
+    /**
+     * @return the content transfer encoding of the upload - this describes how the upload was encoded in the form submission.
+     */
+    String contentTransferEncoding();
```

Comment:
We should not be exposing this, its a protocol level detail that is not really relevant.